src/wp-includes/theme-templates.php

```diff
@@ -1,69 +1,59 @@
 <?php
 
 /**
- * Generates a unique slug for templates.
+ * Finds whether a template slug is customized for the currently active theme.
  *
  * @access private
  * @since 5.8.0
  *
- * @param string $override_slug The filtered value of the slug (starts as `null` from apply_filter).
- * @param string $slug          The original/un-filtered slug (post_name).
- * @param int    $post_ID       Post ID.
- * @param string $post_status   No uniqueness checks are made if the post is still draft or pending.
- * @param string $post_type     Post type.
- * @return string The original, desired slug.
+ * @param string $slug          Template slug.
+ * @param string $template_type wp_template.
+ *
+ * @return bool Whether the template is customized for the currently active theme.
  */
-function wp_filter_wp_template_unique_post_slug( $override_slug, $slug, $post_ID, $post_status, $post_type ) {
-	if ( 'wp_template' !== $post_type ) {
-		return $override_slug;
+function template_is_customized( $slug, $template_type = 'wp_template' ) {
```

Comment:
do you think `block_template_is_customize` is a better name?

Reply:
Maybe, yeah. I was wondering if `*_is_customized` fully captured what this was doing anyway -- maybe `customized_block_template_exists` is more descriptive.